### ISSUE TITLE
TestRedisCommands::test_acl_list fails ... AssertionError: assert 3 == 2

### ISSUE THREAD

**Issue by mcepl** [redis/redis-py#2679]
**Version**: redis-py 4.4.4 and redis 7.0.8

**Platform**: Pythons 3.10.10, 3.11.2, 3.8.16, 3.9.16 on openSUSE/Tumbleweed (development rolling distro)

**Description**: When running the test suite (this crash is with 3.9) I get:

```
[  472s] =================================== FAILURES ===================================
[  472s] _______________________ TestRedisCommands.test_acl_list ________________________
[  472s] 
[  472s] self = <tests.test_commands.TestRedisCommands object at 0x7f77b84d9d00>
[  472s] r = Redis<ConnectionPool<Connection<host=localhost,port=6379,db=9>>>
[  472s] request = <FixtureRequest for <Function test_acl_list>>
[  472s] 
[  472s]     @skip_if_server_version_lt("6.0.0")
[  472s]     @skip_if_redis_enterprise()
[  472s]     def test_acl_list(self, r, request):
[  472s]         username = "redis-py-user"
[  472s]     
[  472s]         def teardown():
[  472s]             r.acl_deluser(username)
[  472s]     
[  472s]         request.addfinalizer(teardown)
[  472s]     
[  472s]         assert r.acl_setuser(username, enabled=False, reset=True)
[  472s]         users = r.acl_list()
[  472s] >       assert len(users) == 2
[  472s] E       AssertionError: assert 3 == 2
[  472s] E        +  where 3 = len(['user default on nopass ~* &* +@all', 'user new_username on #00b9e6622317a2fb628d5514b866d4e7c52b5b149027825645ae3fd72827e84e ~* resetchannels -@all +cluster +command +select +ping +info +flushdb', 'user redis-py-user off sanitize-payload resetchannels -@all'])
```

[Complete build log](https://github.com/redis/redis-py/files/11128279/_log.txt) with all relevant details.

**Comment by dvora-h** [comment]
This issue fixed in version 5.0.0rc2.
@mcepl Can you confirm?

**Comment by mcepl** [comment]
This test passes correctly, but otherwise 5.0.0rc2 has plenty of failing tests I don’t know what to with:

```
[  103s] =================================== FAILURES ===================================
[  103s] __________________ TestRedisCommands.test_acl_getuser_setuser __________________
[  103s] 
[  103s] self = <tests.test_commands.TestRedisCommands object at 0x7f270cbdce80>
[  103s] r = Redis<ConnectionPool<Connection<host=localhost,port=6379,db=0>>>
[  103s] request = <FixtureRequest for <Function test_acl_getuser_setuser>>
[  103s] 
[  103s]     @skip_if_server_version_lt("7.0.0")
[  103s]     @skip_if_redis_enterprise()
[  103s]     def test_acl_getuser_setuser(self, r, request):
[  103s]         r.flushall()
[  103s]         username = "redis-py-user"
[  103s]     
[  103s]         def teardown():
[  103s]             r.acl_deluser(username)
[  103s]     
[  103s]         request.addfinalizer(teardown)
[  103s]     
[  103s]         # test enabled=False
[  103s]         assert r.acl_setuser(username, enabled=False, reset=True)
[  103s]         acl = r.acl_getuser(username)
[  103s]         assert acl["categories"] == ["-@all"]
[  103s]         assert acl["commands"] == []
[  103s]         assert acl["keys"] == []
[  103s]         assert acl["passwords"] == []
[  103s]         assert "off" in acl["flags"]
[  103s]         assert acl["enabled"] is False
[  103s]     
[  103s]         # test nopass=True
[  103s]         assert r.acl_setuser(username, enabled=True, reset=True, nopass=True)
[  103s]         acl = r.acl_getuser(username)
[  103s]         assert acl["categories"] == ["-@all"]
[  103s]         assert acl["commands"] == []
[  103s]         assert acl["keys"] == []
[  103s]         assert acl["passwords"] == []
[  103s]         assert "on" in acl["flags"]
[  103s]         assert "nopass" in acl["flags"]
[  103s]         assert acl["enabled"] is True
[  103s]     
[  103s]         # test all args
[  103s]         assert r.acl_setuser(
[  103s]             username,
[  103s]             enabled=True,
[  103s]             reset=True,
[  103s]             passwords=["+pass1", "+pass2"],
[  103s]             categories=["+set", "+@hash", "-@geo"],
[  103s]             commands=["+get", "+mget", "-hset"],
[  103s]             keys=["cache:*", "objects:*"],
[  103s]         )
[  103s]         acl = r.acl_getuser(username)
[  103s] >       assert set(acl["categories"]) == {"+@hash", "+@set", "-@all", "-@geo"}
[  103s] E       AssertionError: assert {'+@hash', '+@set', '-@all'} == {'+@hash', '+...all', '-@geo'}
[  103s] E         Extra items in the right set:
[  103s] E         '-@geo'
[  103s] E         Full diff:
[  103s] E         - {'-@geo', '+@set', '+@hash', '-@all'}
[  103s] E         ?  ---------
[  103s] E         + {'+@set', '+@hash', '-@all'}
[  103s] 
[  103s] tests/test_commands.py:250: AssertionError
[  103s] _____________________________ test_tcp_ssl_connect _____________________________
[  103s] 
[  103s] tcp_address = ('127.0.0.1', 54925)
[  103s] 
[  103s]     @pytest.mark.ssl
[  103s]     def test_tcp_ssl_connect(tcp_address):
[  103s]         host, port = tcp_address
[  103s] >       certfile = get_ssl_filename("server-cert.pem")
[  103s] 
[  103s] tests/test_connect.py:50: 
[  103s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[  103s] 
[  103s] name = 'server-cert.pem'
[  103s] 
[  103s]     def get_ssl_filename(name):
[  103s]         root = os.path.join(os.path.dirname(__file__), "..")
[  103s]         cert_dir = os.path.abspath(os.path.join(root, "dockers", "stunnel", "keys"))
[  103s]         if not os.path.isdir(cert_dir):  # github actions package validation case
[  103s]             cert_dir = os.path.abspath(
[  103s]                 os.path.join(root, "..", "dockers", "stunnel", "keys")
[  103s]             )
[  103s]             if not os.path.isdir(cert_dir):
[  103s] >               raise IOError(f"No SSL certificates found. They should be in {cert_dir}")
[  103s] E               OSError: No SSL certificates found. They should be in /home/abuild/rpmbuild/BUILD/dockers/stunnel/keys
[  103s] 
[  103s] tests/ssl_utils.py:12: OSError
[  103s] _______ TestRedisCommands.test_acl_getuser_setuser[single-python-parser] _______
[  103s] 
[  103s] self = <tests.test_asyncio.test_commands.TestRedisCommands object at 0x7f270a222490>
[  103s] r_teardown = <function r_teardown.<locals>.factory at 0x7f270a4c3310>
[  103s] 
[  103s]     @skip_if_server_version_lt("7.0.0")
[  103s]     async def test_acl_getuser_setuser(self, r_teardown):
[  103s]         username = "redis-py-user"
[  103s]         r = r_teardown(username)
[  103s]         # test enabled=False
[  103s]         assert await r.acl_setuser(username, enabled=False, reset=True)
[  103s]         acl = await r.acl_getuser(username)
[  103s]         assert acl["categories"] == ["-@all"]
[  103s]         assert acl["commands"] == []
[  103s]         assert acl["keys"] == []
[  103s]         assert acl["passwords"] == []
[  103s]         assert "off" in acl["flags"]
[  103s]         assert acl["enabled"] is False
[  103s]     
[  103s]         # test nopass=True
[  103s]         assert await r.acl_setuser(username, enabled=True, reset=True, nopass=True)
[  103s]         acl = await r.acl_getuser(username)
[  103s]         assert acl["categories"] == ["-@all"]
[  103s]         assert acl["commands"] == []
[  103s]         assert acl["keys"] == []
[  103s]         assert acl["passwords"] == []
[  103s]         assert "on" in acl["flags"]
[  103s]         assert "nopass" in acl["flags"]
[  103s]         assert acl["enabled"] is True
[  103s]     
[  103s]         # test all args
[  103s]         assert await r.acl_setuser(
[  103s]             username,
[  103s]             enabled=True,
[  103s]             reset=True,
[  103s]             passwords=["+pass1", "+pass2"],
[  103s]             categories=["+set", "+@hash", "-geo"],
[  103s]             commands=["+get", "+mget", "-hset"],
[  103s]             keys=["cache:*", "objects:*"],
[  103s]         )
[  103s]         acl = await r.acl_getuser(username)
[  103s] >       assert set(acl["categories"]) == {"-@all", "+@set", "+@hash", "-@geo"}
[  103s] E       AssertionError: assert {'+@hash', '+@set', '-@all'} == {'+@hash', '+...all', '-@geo'}
[  103s] E         Extra items in the right set:
[  103s] E         '-@geo'
[  103s] E         Full diff:
[  103s] E         - {'-@geo', '+@set', '+@hash', '-@all'}
[  103s] E         ?  ---------
[  103s] E         + {'+@set', '+@hash', '-@all'}
[  103s] 
[  103s] tests/test_asyncio/test_commands.py:173: AssertionError
[  103s] ________ TestRedisCommands.test_acl_getuser_setuser[pool-python-parser] ________
[  103s] 
[  103s] self = <tests.test_asyncio.test_commands.TestRedisCommands object at 0x7f270b9985e0>
[  103s] r_teardown = <function r_teardown.<locals>.factory at 0x7f270a4ea160>
[  103s] 
[  103s]     @skip_if_server_version_lt("7.0.0")
[  103s]     async def test_acl_getuser_setuser(self, r_teardown):
[  103s]         username = "redis-py-user"
[  103s]         r = r_teardown(username)
[  103s]         # test enabled=False
[  103s]         assert await r.acl_setuser(username, enabled=False, reset=True)
[  103s]         acl = await r.acl_getuser(username)
[  103s]         assert acl["categories"] == ["-@all"]
[  103s]         assert acl["commands"] == []
[  103s]         assert acl["keys"] == []
[  103s]         assert acl["passwords"] == []
[  103s]         assert "off" in acl["flags"]
[  103s]         assert acl["enabled"] is False
[  103s]     
[  103s]         # test nopass=True
[  103s]         assert await r.acl_setuser(username, enabled=True, reset=True, nopass=True)
[  103s]         acl = await r.acl_getuser(username)
[  103s]         assert acl["categories"] == ["-@all"]
[  103s]         assert acl["commands"] == []
[  103s]         assert acl["keys"] == []
[  103s]         assert acl["passwords"] == []
[  103s]         assert "on" in acl["flags"]
[  103s]         assert "nopass" in acl["flags"]
[  103s]         assert acl["enabled"] is True
[  103s]     
[  103s]         # test all args
[  103s]         assert await r.acl_setuser(
[  103s]             username,
[  103s]             enabled=True,
[  103s]             reset=True,
[  103s]             passwords=["+pass1", "+pass2"],
[  103s]             categories=["+set", "+@hash", "-geo"],
[  103s]             commands=["+get", "+mget", "-hset"],
[  103s]             keys=["cache:*", "objects:*"],
[  103s]         )
[  103s]         acl = await r.acl_getuser(username)
[  103s] >       assert set(acl["categories"]) == {"-@all", "+@set", "+@hash", "-@geo"}
[  103s] E       AssertionError: assert {'+@hash', '+@set', '-@all'} == {'+@hash', '+...all', '-@geo'}
[  103s] E         Extra items in the right set:
[  103s] E         '-@geo'
[  103s] E         Full diff:
[  103s] E         - {'-@geo', '+@set', '+@hash', '-@all'}
[  103s] E         ?  ---------
[  103s] E         + {'+@set', '+@hash', '-@all'}
[  103s] 
[  103s] tests/test_asyncio/test_commands.py:173: AssertionError
[  103s] _____________________________ test_tcp_ssl_connect _____________________________
[  103s] 
[  103s] tcp_address = ('127.0.0.1', 52955)
[  103s] 
[  103s]     @pytest.mark.ssl
[  103s]     async def test_tcp_ssl_connect(tcp_address):
[  103s]         host, port = tcp_address
[  103s] >       certfile = get_ssl_filename("server-cert.pem")
[  103s] 
[  103s] tests/test_asyncio/test_connect.py:55: 
[  103s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[  103s] 
[  103s] name = 'server-cert.pem'
[  103s] 
[  103s]     def get_ssl_filename(name):
[  103s]         root = os.path.join(os.path.dirname(__file__), "..")
[  103s]         cert_dir = os.path.abspath(os.path.join(root, "dockers", "stunnel", "keys"))
[  103s]         if not os.path.isdir(cert_dir):  # github actions package validation case
[  103s]             cert_dir = os.path.abspath(
[  103s]                 os.path.join(root, "..", "dockers", "stunnel", "keys")
[  103s]             )
[  103s]             if not os.path.isdir(cert_dir):
[  103s] >               raise IOError(f"No SSL certificates found. They should be in {cert_dir}")
[  103s] E               OSError: No SSL certificates found. They should be in /home/abuild/rpmbuild/BUILD/dockers/stunnel/keys
[  103s] 
[  103s] tests/ssl_utils.py:12: OSError
[  103s] =============================== warnings summary ===============================
[  103s] tests/test_asyncio/test_connection.py::test_connection_disconect_race[AsyncRESP2Parser]
[  103s] tests/test_asyncio/test_connection.py::test_connection_disconect_race[AsyncRESP3Parser]
[  103s]   /home/abuild/rpmbuild/BUILD/redis-5.0.0rc2/redis/asyncio/connection.py:365: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
[  103s]     self._writer.close()  # type: ignore[union-attr]
[  103s]   Enable tracemalloc to get traceback where the object was allocated.
[  103s]   See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
[  103s] 
[  103s] tests/test_asyncio/test_cwe_404.py::test_standalone[0.5]
[  103s]   /usr/lib/python3.9/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <coroutine object DelayProxy.handle at 0x7f270c2a60c0>
[  103s]   
[  103s]   Traceback (most recent call last):
[  103s]     File "/home/abuild/rpmbuild/BUILD/redis-5.0.0rc2/tests/test_asyncio/test_cwe_404.py", line 57, in handle
[  103s]       await asyncio.gather(pipe1, pipe2)
[  103s]   GeneratorExit
[  103s]   
[  103s]   During handling of the above exception, another exception occurred:
[  103s]   
[  103s]   Traceback (most recent call last):
[  103s]     File "/home/abuild/rpmbuild/BUILD/redis-5.0.0rc2/tests/test_asyncio/test_cwe_404.py", line 59, in handle
[  103s]       redis_writer.close()
[  103s]     File "/usr/lib64/python3.9/asyncio/streams.py", line 353, in close
[  103s]       return self._transport.close()
[  103s]     File "/usr/lib64/python3.9/asyncio/selector_events.py", line 698, in close
[  103s]       self._loop.call_soon(self._call_connection_lost, None)
[  103s]     File "/usr/lib64/python3.9/asyncio/base_events.py", line 751, in call_soon
[  103s]       self._check_closed()
[  103s]     File "/usr/lib64/python3.9/asyncio/base_events.py", line 515, in _check_closed
[  103s]       raise RuntimeError('Event loop is closed')
[  103s]   RuntimeError: Event loop is closed
[  103s]   
[  103s]     warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
[  103s] 
[  103s] tests/test_asyncio/test_cwe_404.py::test_standalone[2]
[  103s]   /usr/lib/python3.9/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <coroutine object DelayProxy.handle at 0x7f270bdca740>
[  103s]   
[  103s]   Traceback (most recent call last):
[  103s]     File "/home/abuild/rpmbuild/BUILD/redis-5.0.0rc2/tests/test_asyncio/test_cwe_404.py", line 57, in handle
[  103s]       await asyncio.gather(pipe1, pipe2)
[  103s]   GeneratorExit
[  103s]   
[  103s]   During handling of the above exception, another exception occurred:
[  103s]   
[  103s]   Traceback (most recent call last):
[  103s]     File "/home/abuild/rpmbuild/BUILD/redis-5.0.0rc2/tests/test_asyncio/test_cwe_404.py", line 59, in handle
[  103s]       redis_writer.close()
[  103s]     File "/usr/lib64/python3.9/asyncio/streams.py", line 353, in close
[  103s]       return self._transport.close()
[  103s]     File "/usr/lib64/python3.9/asyncio/selector_events.py", line 698, in close
[  103s]       self._loop.call_soon(self._call_connection_lost, None)
[  103s]     File "/usr/lib64/python3.9/asyncio/base_events.py", line 751, in call_soon
[  103s]       self._check_closed()
[  103s]     File "/usr/lib64/python3.9/asyncio/base_events.py", line 515, in _check_closed
[  103s]       raise RuntimeError('Event loop is closed')
[  103s]   RuntimeError: Event loop is closed
[  103s]   
[  103s]     warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
[  103s] 
[  103s] tests/test_asyncio/test_cwe_404.py::test_standalone_pipeline[0.5]
[  103s]   /usr/lib/python3.9/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <coroutine object DelayProxy.handle at 0x7f270b55d440>
[  103s]   
[  103s]   Traceback (most recent call last):
[  103s]     File "/home/abuild/rpmbuild/BUILD/redis-5.0.0rc2/redis/asyncio/connection.py", line 1075, in get_connection
[  103s]       connection = self._available_connections.pop()
[  103s]   IndexError: pop from empty list
[  103s]   
[  103s]   During handling of the above exception, another exception occurred:
[  103s]   
[  103s]   Traceback (most recent call last):
[  103s]     File "/home/abuild/rpmbuild/BUILD/redis-5.0.0rc2/tests/test_asyncio/test_cwe_404.py", line 57, in handle
[  103s]       await asyncio.gather(pipe1, pipe2)
[  103s]   GeneratorExit
[  103s]   
[  103s]   During handling of the above exception, another exception occurred:
[  103s]   
[  103s]   Traceback (most recent call last):
[  103s]     File "/home/abuild/rpmbuild/BUILD/redis-5.0.0rc2/tests/test_asyncio/test_cwe_404.py", line 59, in handle
[  103s]       redis_writer.close()
[  103s]     File "/usr/lib64/python3.9/asyncio/streams.py", line 353, in close
[  103s]       return self._transport.close()
[  103s]     File "/usr/lib64/python3.9/asyncio/selector_events.py", line 698, in close
[  103s]       self._loop.call_soon(self._call_connection_lost, None)
[  103s]     File "/usr/lib64/python3.9/asyncio/base_events.py", line 751, in call_soon
[  103s]       self._check_closed()
[  103s]     File "/usr/lib64/python3.9/asyncio/base_events.py", line 515, in _check_closed
[  103s]       raise RuntimeError('Event loop is closed')
[  103s]   RuntimeError: Event loop is closed
[  103s]   
[  103s]     warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
[  103s] 
[  103s] tests/test_asyncio/test_cwe_404.py::test_standalone_pipeline[2]
[  103s]   /usr/lib/python3.9/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <coroutine object DelayProxy.handle at 0x7f270c32f440>
[  103s]   
[  103s]   Traceback (most recent call last):
[  103s]     File "/home/abuild/rpmbuild/BUILD/redis-5.0.0rc2/tests/test_asyncio/test_cwe_404.py", line 57, in handle
[  103s]       await asyncio.gather(pipe1, pipe2)
[  103s]   GeneratorExit
[  103s]   
[  103s]   During handling of the above exception, another exception occurred:
[  103s]   
[  103s]   Traceback (most recent call last):
[  103s]     File "/home/abuild/rpmbuild/BUILD/redis-5.0.0rc2/tests/test_asyncio/test_cwe_404.py", line 59, in handle
[  103s]       redis_writer.close()
[  103s]     File "/usr/lib64/python3.9/asyncio/streams.py", line 353, in close
[  103s]       return self._transport.close()
[  103s]     File "/usr/lib64/python3.9/asyncio/selector_events.py", line 698, in close
[  103s]       self._loop.call_soon(self._call_connection_lost, None)
[  103s]     File "/usr/lib64/python3.9/asyncio/base_events.py", line 751, in call_soon
[  103s]       self._check_closed()
[  103s]     File "/usr/lib64/python3.9/asyncio/base_events.py", line 515, in _check_closed
[  103s]       raise RuntimeError('Event loop is closed')
[  103s]   RuntimeError: Event loop is closed
[  103s]   
[  103s]     warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
[  103s] 
[  103s] tests/test_asyncio/test_encoding.py::TestEncoding::test_memoryview_encoding[pool-python-parser]
[  103s]   /usr/lib/python3.9/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <coroutine object DelayProxy.handle at 0x7f270bdd9b40>
[  103s]   
[  103s]   Traceback (most recent call last):
[  103s]     File "/home/abuild/rpmbuild/BUILD/redis-5.0.0rc2/tests/test_asyncio/test_cwe_404.py", line 57, in handle
[  103s]       await asyncio.gather(pipe1, pipe2)
[  103s]   GeneratorExit
[  103s]   
[  103s]   During handling of the above exception, another exception occurred:
[  103s]   
[  103s]   Traceback (most recent call last):
[  103s]     File "/home/abuild/rpmbuild/BUILD/redis-5.0.0rc2/tests/test_asyncio/test_cwe_404.py", line 59, in handle
[  103s]       redis_writer.close()
[  103s]     File "/usr/lib64/python3.9/asyncio/streams.py", line 353, in close
[  103s]       return self._transport.close()
[  103s]     File "/usr/lib64/python3.9/asyncio/selector_events.py", line 698, in close
[  103s]       self._loop.call_soon(self._call_connection_lost, None)
[  103s]     File "/usr/lib64/python3.9/asyncio/base_events.py", line 751, in call_soon
[  103s]       self._check_closed()
[  103s]     File "/usr/lib64/python3.9/asyncio/base_events.py", line 515, in _check_closed
[  103s]       raise RuntimeError('Event loop is closed')
[  103s]   RuntimeError: Event loop is closed
[  103s]   
[  103s]     warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
[  103s] 
[  103s] tests/test_asyncio/test_encoding.py::TestCommandsAreNotEncoded::test_basic_command[single-python-parser]
[  103s]   /usr/lib/python3.9/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <coroutine object DelayProxy.handle at 0x7f270bdd4340>
[  103s]   
[  103s]   Traceback (most recent call last):
[  103s]     File "/home/abuild/rpmbuild/BUILD/redis-5.0.0rc2/tests/test_asyncio/test_cwe_404.py", line 57, in handle
[  103s]       await asyncio.gather(pipe1, pipe2)
[  103s]   GeneratorExit
[  103s]   
[  103s]   During handling of the above exception, another exception occurred:
[  103s]   
[  103s]   Traceback (most recent call last):
[  103s]     File "/home/abuild/rpmbuild/BUILD/redis-5.0.0rc2/tests/test_asyncio/test_cwe_404.py", line 59, in handle
[  103s]       redis_writer.close()
[  103s]     File "/usr/lib64/python3.9/asyncio/streams.py", line 353, in close
[  103s]       return self._transport.close()
[  103s]     File "/usr/lib64/python3.9/asyncio/selector_events.py", line 698, in close
[  103s]       self._loop.call_soon(self._call_connection_lost, None)
[  103s]     File "/usr/lib64/python3.9/asyncio/base_events.py", line 751, in call_soon
[  103s]       self._check_closed()
[  103s]     File "/usr/lib64/python3.9/asyncio/base_events.py", line 515, in _check_closed
[  103s]       raise RuntimeError('Event loop is closed')
[  103s]   RuntimeError: Event loop is closed
[  103s]   
[  103s]     warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
[  103s] 
[  103s] -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
[  103s] =========================== short test summary info ============================
[  103s] FAILED tests/test_commands.py::TestRedisCommands::test_acl_getuser_setuser - ...
[  103s] FAILED tests/test_connect.py::test_tcp_ssl_connect - OSError: No SSL certific...
[  103s] FAILED tests/test_asyncio/test_commands.py::TestRedisCommands::test_acl_getuser_setuser[single-python-parser]
[  103s] FAILED tests/test_asyncio/test_commands.py::TestRedisCommands::test_acl_getuser_setuser[pool-python-parser]
[  103s] FAILED tests/test_asyncio/test_connect.py::test_tcp_ssl_connect - OSError: No...
[  103s] = 5 failed, 1783 passed, 940 skipped, 969 deselected, 22 xpassed, 8 warnings in 84.90s (0:01:24) =
[  103s] + kill 16514 16520
[  103s] 16520:signal-handler (1689693902) Received SIGTERM scheduling shutdown...
[  103s] 16514:signal-handler (1689693902) Received SIGTERM scheduling shutdown...
```
[Complete build log](https://github.com/redis/redis-py/files/12082973/_log.txt)
